winpr/libwinpr/crt/unicode.c

```diff
@@ -374,12 +374,12 @@ void ByteSwapUnicode(WCHAR* wstr, size_t length)
 SSIZE_T ConvertWCharToUtf8(const WCHAR* wstr, char* str, size_t len)
 {
 	if (!wstr)
-		return 0;
+		return -1;
```

Comment:
this change is wrong, a `NULL` wstr` is allowed and should not return an error.

Reply:
My bad did not tilt when reading the API doc, I'll revert this.

I disagree on the fact that NULL should be allowed as an input.
You won't be able to make a difference between NULL and an empty string.
So if you have a NULL WCHAR*, first call will return 0.
So you allocate a buffer of size 1 for the null termination.
Then you call the function again and nothing appens.
In the end you may have an uninitialiazed allocated buffer of size 1.
Not the worst thing in the world, but not the cleanest in my oppinion.

Reply:
@DVeron-RC you are right about the uninitialized part, should initialize the buffer, so if you could add that would be great ;)

Reply:
On it.

The more I think about it the more I still think NULL should not be allowed.
I don't understand why in ConvertWCharToUtf8 a NULL WCHAR* is valid,
whereas in ConvertWCharNToUtf8 it is not.

How does freerdp update its API ?

Reply:
Ok, I'm dumb in the N version str can be NULL if len is 0. Coherent.